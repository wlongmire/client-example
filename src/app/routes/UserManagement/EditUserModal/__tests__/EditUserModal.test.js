/* eslint-env node, jest */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { EditUserModal } from '../index'

const loggedInUser = {
  brokerId: 'test-2222-7fd-b3ff-4fd3-9fc2-e752b9f5b002',
  dateCreated: '2017-10-11T18:41:14.000Z',
  email: 'allison.e.steranko@gmail.com',
  id: 'asdasdasd-08db-4c32-aa38-500808819b44',
  lastOnline: 'N/A',
  role: 'admin',
  status: 'active',
  username: 'asdasdas.e.asdasd@gmail.com'
}


const selectedUser = {
  brokerId: 'test-123123-7fd-b3ff-4fd3-9fc2-e752b9f5b002',
  dateCreated: '2017-10-11T18:41:14.000Z',
  email: 'andriy123@gmail.com',
  id: '1123-08db-4c32-aa38-213124',
  lastOnline: 'N/A',
  role: 'admin',
  status: 'pending',
  username: 'asdasdasd.e.asdasd@gmail.com'
}

const selectedUser2 = {
  brokerId: 'test-123123-7fd-b3ff-4fd3-9fc2-e752b9f5b002',
  dateCreated: '2017-10-11T18:41:14.000Z',
  email: 'andriy123@gmail.com',
  id: '1123-08db-4c32-aa38-213124',
  lastOnline: 'N/A',
  role: 'admin',
  status: 'pending',
  username: 'asdasdasd.e.asdasd@gmail.com'
}

describe('>>> EditUserModal Component (Admin)--- Snapshot', () => {
  it('+++capturing Snapshot with', () => {
    const renderedValue = shallow(
      <EditUserModal
        selectedUser={selectedUser}
        loggedInUser={loggedInUser}
        hideEditModal={jest.fn()}
      />)
    expect(toJson(renderedValue)).toMatchSnapshot()
  })
})

describe('>>> EditUserModal Component (Non - Admin)--- Snapshot', () => {
  it('+++capturing Snapshot with', () => {
    const renderedValue = shallow(
      <EditUserModal
        selectedUser={selectedUser2}
        loggedInUser={loggedInUser}
        hideEditModal={jest.fn()}
      />)
    expect(toJson(renderedValue)).toMatchSnapshot()
  })
})