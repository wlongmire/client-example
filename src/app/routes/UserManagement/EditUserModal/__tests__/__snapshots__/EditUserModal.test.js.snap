// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> EditUserModal Component (Admin)--- Snapshot +++capturing Snapshot with 1`] = `
<div
  className="editUserModal"
>
  <h3>
    Email:
  </h3>
  <h4>
    andriy123@gmail.com
  </h4>
  <h3
    className="editAminText"
  >
    Is user an admin?
  </h3>
  <form
    onSubmit={[Function]}
  >
    <FormGroup
      bsClass="form-group"
    >
      <br />
      <Radio
        bsClass="radio"
        checked={true}
        className="editAdminInput"
        disabled={false}
        inline={true}
        name="radioGroup"
        onChange={[Function]}
        required={true}
        value="true"
      >
        Yes
      </Radio>
       
      <Radio
        bsClass="radio"
        checked={false}
        className="editAdminInput"
        disabled={false}
        inline={true}
        name="radioGroup"
        onChange={[Function]}
        required={true}
        value="false"
      >
        No
      </Radio>
    </FormGroup>
    <p
      className="adminEditAddInfo"
    >
      Admins can add and remove other users.
    </p>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="saveEditChanges"
      disabled={false}
      type="submit"
    >
      Save Changes
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="cancelEdit"
      disabled={false}
      onClick={[Function]}
    >
      Cancel
    </Button>
  </form>
</div>
`;

exports[`>>> EditUserModal Component (Non - Admin)--- Snapshot +++capturing Snapshot with 1`] = `
<div
  className="editUserModal"
>
  <h3>
    Email:
  </h3>
  <h4>
    andriy123@gmail.com
  </h4>
  <h3
    className="editAminText"
  >
    Is user an admin?
  </h3>
  <form
    onSubmit={[Function]}
  >
    <FormGroup
      bsClass="form-group"
    >
      <br />
      <Radio
        bsClass="radio"
        checked={true}
        className="editAdminInput"
        disabled={false}
        inline={true}
        name="radioGroup"
        onChange={[Function]}
        required={true}
        value="true"
      >
        Yes
      </Radio>
       
      <Radio
        bsClass="radio"
        checked={false}
        className="editAdminInput"
        disabled={false}
        inline={true}
        name="radioGroup"
        onChange={[Function]}
        required={true}
        value="false"
      >
        No
      </Radio>
    </FormGroup>
    <p
      className="adminEditAddInfo"
    >
      Admins can add and remove other users.
    </p>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="saveEditChanges"
      disabled={false}
      type="submit"
    >
      Save Changes
    </Button>
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsStyle="default"
      className="cancelEdit"
      disabled={false}
      onClick={[Function]}
    >
      Cancel
    </Button>
  </form>
</div>
`;
